### PR TITLE
Allow for negative counter values

### PR DESCRIPTION
**Pull request by pirgeo**

### CODE REVIEW

File: metric/metric_test.go [hunk 1]
```diff
@@ -297,36 +297,32 @@ func TestNewMetric(t *testing.T) {
...
 			}},
-			wantErr: true,
-			want:    nil,
+			want: &Metric{metricKey: "name", value: intCounterValue{-3, false}},
 		},
 		{
-			name: "invalid absolute int counter value",
+			name: "negative absolute int counter value",
 			args: args{metricKey: "name", options: []MetricOption{
 				WithIntCounterValueDelta(-3),
```
Comment (arminru) [inline metric/metric_test.go:309]: While you're at it you could rename the test names to match the API.

**Review by dyladan — APPROVED** [review]
Looks good +/- Armin's suggested test naming change.

**Review by arminru — APPROVED** [review]

**Comment by dyladan** [comment]
There is no version constant to change in the go library right?

**Comment by pirgeo** [comment]
I dont think so, I think Go does the versioning via tags, right?

**Comment by dyladan** [comment]
yes but in some projects we had constants anyway so the library could use it in things like user agent. Couldn't remember if this was one of those projects.

**Comment by pirgeo** [comment]
Oh, i see. This project doesn't have that, thanks for bringing it up. I usually always forget the version number changes :roll_eyes:

**Pull request merged**